internal/entity/face.go

```diff
@@ -273,9 +273,14 @@ func (m *Face) RefreshPhotos() error {
 		return fmt.Errorf("empty face id")
 	}
 
-	update := fmt.Sprintf(
-		"UPDATE photos SET checked_at = NULL WHERE id IN (SELECT f.photo_id FROM files f JOIN %s m ON m.file_uid = f.file_uid WHERE m.face_id = ?)",
-		Marker{}.TableName())
+	var update string
+	switch DbDialect() {
+	case MySQL:
+		update = "UPDATE photos p JOIN files f ON f.photo_id = p.id JOIN %s m ON m.file_uid = f.file_uid SET p.checked_at = NULL WHERE m.face_id = ?"
+	default:
+		update = "UPDATE photos SET checked_at = NULL WHERE id IN (SELECT f.photo_id FROM files f JOIN %s m ON m.file_uid = f.file_uid WHERE m.face_id = ?)"
+	}
+	update = fmt.Sprintf(update, Marker{}.TableName())
```

Comment:
May lead to problems in that dialects can have different parameters or need them in a different order. So it's much easier for developers to make mistakes or they have to refactor backwards.

Reply:
You're right, that's an unnecessary footgun. I'll pull the `.Exec()` call up into the branches.
Related: I was wondering why the `markers` table name is pulled from the according class via `.TableName()`, but the others aren't.

Reply:
Because we used temporary table names during development, while the other table names did not change after the initial stable release.

Reply:
Updated the PR. Tests still pass. Wasn't sure about the err var, could have put return statements into the branches, but I think it can be confusing if a non-void function doesn't end in a return statement.